### DISCUSSION TITLE
Please help me: I encountered an error when recompiling the source code that I could not find the header file

### DISCUSSION THREAD

**Discussion by Jsupermaster** [B-Lang-org/bsc#741] [Q&A] [answered]
When I execute make install-src, I get the following error message indicating that the header file cannot be found. My running platform is the wsl ubuntu24.04 virtual machine on windows11 for ARM platform

make  -C src  PREFIX=/opt/tools/bsc/bsc/inst  install
make[1]: Entering directory '/opt/tools/bsc/bsc/src'
make  -C vendor/stp   PREFIX=/opt/tools/bsc/bsc/inst  install
make[2]: Entering directory '/opt/tools/bsc/bsc/src/vendor/stp'
make -C src install
make[3]: Entering directory '/opt/tools/bsc/bsc/src/vendor/stp/src'
make -C AST
make[4]: Entering directory '/opt/tools/bsc/bsc/src/vendor/stp/src/AST'
Making dependencies
/bin/sh: 1: cannot create depend: Permission denied
g++ -O3 -DNDEBUG -fomit-frame-pointer -fPIC  -D __STDC_LIMIT_MACROS -D __STDC_FORMAT_MACROS -O3 -DNDEBUG -fomit-frame-pointer -fPIC  -D __STDC_LIMIT_MACROS -D __STDC_FORMAT_MACROS -Wall -std=c++11 -DTR1_UNORDERED_MAP -Wno-deprecated   -c -o ASTBVConst.o ASTBVConst.cpp
In file included from AST.h:12,
                 from ASTBVConst.cpp:10:
UsefulDefs.h:25:10: fatal error: ASTKind.h: No such file or directory
   25 | #include "ASTKind.h"
      |          ^~~~~~~~~~~
compilation terminated.
make[4]: *** [<builtin>: ASTBVConst.o] Error 1
make[4]: Leaving directory '/opt/tools/bsc/bsc/src/vendor/stp/src/AST'
make[3]: *** [Makefile:101: AST/libast.a] Error 2
make[3]: Leaving directory '/opt/tools/bsc/bsc/src/vendor/stp/src'
make[2]: *** [Makefile:23: install] Error 2
make[2]: Leaving directory '/opt/tools/bsc/bsc/src/vendor/stp'
make[1]: *** [Makefile:57: install] Error 2
make[1]: Leaving directory '/opt/tools/bsc/bsc/src'
make: *** [GNUmakefile:41: install-src] Error 2

Please help me!

**Comment by quark17** [comment, marked as answer]
The problem is a few lines earlier in the output:
```
/bin/sh: 1: cannot create depend: Permission denied
```
It could be that you don't have write permission to the `bsc` directories.

**Reply by Jsupermaster** [reply]
Thank you for your answer. This question is a stupid one caused by my carelessness.
